llvm/unittests/ADT/APFloatTest.cpp

```diff
@@ -1378,6 +1378,171 @@ TEST(APFloatTest, toString) {
   }
 }
 
+TEST(APFloatTest, format) {
+  auto doit_precision = [](const fltSemantics &S, const char *value,
+                           llvm::FloatStyle style, int precision,
+                           const char *expected) {
+    SmallString<16> s;
+    APFloat apf(S, value);
+    apf.format(s, style, precision);
+    EXPECT_STREQ(expected, s.c_str());
```

Comment:
```suggestion
    EXPECT_EQ(expected, s);
```

Regular EXPECT_EQ should work fine with StringRef + SmallString